ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KtlintSuppressionRuleTest.kt

```diff
@@ -0,0 +1,798 @@
+            val code =
+                """
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:Suppress("ktlint:standard:no-wildcard-imports")
+
+                import foo.bar
+                import foobar.*
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(2, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given an EOL comment with a ktlint-disable directive on an import and an existing @file Suppress annotation`() {
+            val code =
+                """
+                @file:Suppress("aaa", "zzz")
+
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
+                """.trimIndent()
+            val formattedCode =
+                """
+                @file:Suppress("aaa", "ktlint:standard:no-wildcard-imports", "zzz")
+
+                import foo.bar
+                import foobar.*
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(4, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given an EOL comment with a ktlint-disable directive on an import on a file starting with a (copyright) comment before the package statement`() {
+            val code =
+                """
+                /* Some copyright notice before package statement */
+                package foo
+
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
+                """.trimIndent()
+            val formattedCode =
+                """
+                /* Some copyright notice before package statement */
+                @file:Suppress("ktlint:standard:no-wildcard-imports")
+
+                package foo
+
+                import foo.bar
+                import foobar.*
+                """.trimIndent()
+            ktlintSuppressionRuleAssertThat(code)
+                .hasLintViolation(5, 20, "Directive 'ktlint-disable' is deprecated. Replace with @Suppress annotation")
+                .isFormattedAs(formattedCode)
+        }
+
+        @Test
+        fun `Given an EOL comment with a ktlint-disable directive on an import and an existing @file Suppress annotation  on a file starting with a (copyright) comment`() {
+            val code =
+                """
+                /* Some copyright notice before package statement */
+                @file:Suppress("aaa", "zzz")
+                package foo
+
+                import foo.bar
+                import foobar.* // ktlint-disable no-wildcard-imports
```

Comment:
Does it make sense to add this TC `import foobar.* // ktlint-disable no-wildcard-imports // some comment`?

Reply:
Lol, you're really creative. The SuppressionLocatorBuilder will print warnings but processes it anyways. i am not in favor to try to be 100% fool proof. I expect a user to inspect format fixes proposed by ktlint before committing.